src/schema/formats.js

```diff
@@ -1,48 +1,27 @@
 import { URL } from 'whatwg-url';
 
-const VALIDNUMRX = /^[0-9]{1,5}$/;
-
-// Firefox's version format is laxer than Chrome's, it accepts:
-// https://developer.mozilla.org/en-US/docs/Toolkit_version_format
-// We choose a slightly restricted version of that format (but still more
-// permissive than Chrome) to allow Beta addons, per:
-// https://developer.mozilla.org/en-US/Add-ons/AMO/Policy/Maintenance
-const TOOLKIT_VERSION_REGEX = /^(\d+\.?){1,3}\.(\d+([A-z]+(-?[\dA-z]+)?))$/;
-
 export function isValidVersionString(version) {
   // We should be starting with a string.
-  if (typeof version !== 'string') {
-    return false;
-  }
-  // If valid toolkit version string, return true early
-  if (TOOLKIT_VERSION_REGEX.test(version)) {
-    return true;
-  }
-  const parts = version.split('.');
-  if (parts.length > 4) {
+  if (!(typeof version === 'string' || version instanceof String)) {
     return false;
   }
-
-  for (let i = 0; i < parts.length; i++) {
-    let part = parts[i];
-    // Leading or multiple zeros not allowed.
-    if (part.startsWith('0') && part.length > 1) {
-      return false;
-    }
-    // Disallow things like 123e5 which parseInt will convert.
-    if (!VALIDNUMRX.test(part)) {
-      return false;
-    }
-    part = parseInt(part, 10);
-    if (Number.isNaN(part) || part < 0 || part > 65535) {
-      return false;
-    }
-  }
-  return true;
+  // Firefox's version format is laxer than Chrome's, it accepts:
+  // https://developer.mozilla.org/en-US/docs/Toolkit_version_format
+  // We choose a slightly restricted version of that format (but still more
+  // permissive than Chrome) to allow Beta addons, per:
+  // https://developer.mozilla.org/en-US/Add-ons/AMO/Policy/Maintenance
+  // We also support Semantic Versioning:
+  // http://semver.org/
+  const chromePart = '(?:0|[1-9]\\d{0,3}|[1-5]\\d{4}|6(?:[0-4]\\d{3}|5(?:[0-4]\\d{2}|5(?:[0-2]\\d|3[0-5]))))';
+  const chromeVer = `${chromePart}(?:\\.${chromePart}){0,3}`;
+  const betaPart = '(?:a(?:lpha)?|b(?:eta)?|pre|rc)\\d*';
+  const addonVer = `${chromeVer}(?:${betaPart})?`;
+  const semVer = '(?:0|[1-9]\\d*)(?:\\.(?:0|[1-9]\\d*)){2}(?:-[A-Za-z\\d\\-]+(?:\\.[A-Za-z\\d\\-]+)*)?(?:\\+[A-Za-z\\d\\-]+(?:\\.[A-Za-z\\d\\-]+)*)?';
+  return (new RegExp(`^(?:${addonVer}|${semVer})$`)).test(version);
```

Comment:
I think those regular expressions should use what Firefox accepts, not more and not less.

Reply:
I am not a native English speaker, so I can not understand a bit of the meaning of what you are pointing out.
What does "use what Firefox accepts" mean?

Reply:
We should use the format that Firefox supports.

Reply:
You mean only Toolkit?

Reply:
Chrome version format always matches Toolkit. (&lt;number-a&gt;)
SemVer should match Toolkit too.
For example, `1.0.0-a1b1`.
According to [Toolkit version format | MDN](https://developer.mozilla.org/en-US/docs/Toolkit_version_format "Toolkit version format | MDN"), this version looks like it doesn't match Toolkit version format.
In 3rd version part (`0-a1b1`), &lt;number-a&gt; is `0` &lt;string-b&gt; is `-a` &lt;number-c&gt; is `1` &lt;string-d&gt; is `b`, last `1` does not match.
But according to [nsIVersionComparator - Mozilla | MDN](https://developer.mozilla.org/en-US/docs/Mozilla/Tech/XPCOM/Reference/Interface/nsIVersionComparator "nsIVersionComparator - Mozilla | MDN"), it says
&gt; &lt;number-a&gt;&lt;string-b&gt;&lt;number-c&gt;&lt;string-d (everything else)&gt;
If so, then &lt;string-d&gt; becomes `b1`, and it matches Toolkit?